tests/parser/functions/test_slice.py

```diff
@@ -146,15 +152,19 @@ def _get_contract():
     if (
         (start + length > data_length and literal_start and literal_length)
         or (literal_length and length > data_length)
-        or (location == "literal" and len(bytesdata) > length_bound)
+        or (
+            location == "literal"
+            and len(bytesdata) > length_bound
+            and ((literal_length and length > length_bound) or (not literal_length))
+        )
         or (literal_start and start > data_length)
         or (literal_length and length < 1)
     ):
         assert_compile_failed(lambda: _get_contract(), (ArgumentException, TypeMismatch))
-    elif len(bytesdata) > data_length:
+    elif location in "code" and len(bytesdata) > data_length:
```

Comment:
this test looks a bit sus. does this ever evaluate to true?

Reply:
Yes, e.g. `location="code", bytesdata=b"\x00\x00", ..., literal_start=False, ..., literal_length=False, length_bound=1`

```
IMMUTABLE_BYTES: immutable(Bytes[1])
@external
def __init__(foo: Bytes[1]):
    IMMUTABLE_BYTES = foo

@external
def do_slice(inp: Bytes[1], start: uint256, length: uint256) -> Bytes[1]:
    return slice(IMMUTABLE_BYTES, start, length)
```

Reply:
oh oops I see what the issue is now :see_no_evil: